### PR TITLE
BTA-3279

### PR DESCRIPTION
**Pull request by HarryGull**
See also data fixes for this:
https://github.com/hmrc/view-vat-acceptance-tests/pull/93

### CODE REVIEW

File: app/connectors/VatObligationsConnector.scala [hunk 1]
```diff
@@ -42,7 +42,9 @@ class VatObligationsConnector @Inject()(val http: HttpClient, val appConfig: Mic
 
     Logger.debug(s"[VatObligationsConnector][getVatObligations] - Calling GET $url \nHeaders: $desHC\n QueryParams: $queryParameters")
     http.GET(url, queryParameters.toSeqQueryParams)(VatObligationsReads, desHC, ec).map {
-      case vatObligations@Right(_) => vatObligations
+      case vatObligations@Right(_) => {
```
Comment (lewishay) [inline app/connectors/VatObligationsConnector.scala]: Redundant curly braces here (IntelliJ gives a warning).

File: test/connectors/httpParsers/VatObligationsHttpParserSpec.scala [hunk 2]
```diff
@@ -80,6 +79,114 @@ class VatObligationsHttpParserSpec extends SpecBase {
...
+
+      val testObligations: VatObligations =
+        VatObligations(
+          Seq.empty[VatObligation]
+          )
+
+
+      val responseJson: JsValue = Json.parse(
+        """{
+          |    "obligations": []
```
Comment (lewishay) [inline test/connectors/httpParsers/VatObligationsHttpParserSpec.scala]: The `obligationDetails` field inside of `obligations` is required by DES so this technically does not match the schema.

File: test/connectors/httpParsers/VatObligationsHttpParserSpec.scala [hunk 3]
```diff
@@ -80,6 +79,114 @@ class VatObligationsHttpParserSpec extends SpecBase {
...
+      val expected: Either[Nothing, VatObligations] = Right(testObligations)
+      val result: VatObligationsHttpParser.HttpGetResult[VatObligations] = VatObligationsReads.read("", "", httpResponse)
+
+      "return a VatObligations instance" in {
+        result shouldEqual expected
+      }
+    }
+
+
+    "response status is 200 OK and matches expected Schema when the response contains multiple obligations with multiple details" should {
```
Comment (lewishay) [inline test/connectors/httpParsers/VatObligationsHttpParserSpec.scala]: Is this a valid test? When would the API send back multiple `obligationDetails`?

File: test/controllers/VatObligationsControllerSpec.scala [hunk 4]
```diff
@@ -30,18 +30,35 @@ class VatObligationsControllerSpec extends SpecBase with MockVatObligationsServi
...
       Seq(VatObligation(
-        ObligationIdentification("555555555", "VRN"),
         Seq(
           ObligationDetail("F", "1980-02-03", "1980-04-05", Some("1980-02-02"), "1980-04-08", "17AA"),
           ObligationDetail("F", "1981-02-03", "1981-04-05", Some("1981-02-02"), "1981-04-08", "18AA")
         )
       )
       ))
 
+  val successMultipleObligations: VatObligations =
```
Comment (lewishay) [inline test/controllers/VatObligationsControllerSpec.scala:40]: Same question here as in the HttpParserSpec - is multiple arrays of obligations a valid scenario?

File: test/models/VatObligationsSpec.scala [hunk 5]
```diff
@@ -21,56 +21,208 @@ import play.api.libs.json.{JsValue, Json}
...
-    VatObligations(
-      Seq(VatObligation(
-        ObligationIdentification("555555555", "VRN"),
-        Seq(
-          ObligationDetail("F", "1980-02-03", "1980-04-05", Some("1980-02-02"), "1980-04-08", "17AA"),
-          ObligationDetail("F", "1981-02-03", "1981-04-05", Some("1981-02-02"), "1981-04-08", "18AA")
+
+  "VatObligations when there is a single obligation" should {
+
+    val singleObligationInputJson: JsValue = Json.parse(
```
Comment (lewishay) [inline test/models/VatObligationsSpec.scala]: This test says "when there is a single obligation" but there are multiple obligations in this JSON.

File: test/models/VatObligationsSpec.scala [hunk 6]
```diff
@@ -21,56 +21,208 @@ import play.api.libs.json.{JsValue, Json}
...
-      vatObligationsJson.as[VatObligations] shouldBe vatObligationsModel
+      singleObligationInputJson.as[VatObligations] shouldBe singleObligationsModel
     }
 
   }
 
+
+  "VatObligations with multiple obligations" should {
+
+    val multipleObligationInputJson: JsValue = Json.parse(
```
Comment (lewishay) [inline test/models/VatObligationsSpec.scala]: Same comment here as in the HttpParserSpec and ControllerSpec regarding valid scenario.

**Review by lewishay — CHANGES_REQUESTED** [review]
Could you also please squash your commits to remove the "workInProgress" commit message just to keep the history clean.

File: app/connectors/VatObligationsConnector.scala (hunk 1 above)
Comment (HarryGull) [reply app/connectors/VatObligationsConnector.scala]: fixed

File: test/connectors/httpParsers/VatObligationsHttpParserSpec.scala (hunk 2 above)
Comment (HarryGull) [reply test/connectors/httpParsers/VatObligationsHttpParserSpec.scala]: This is the test that led to me saying it should return a 404 and correcting alll the data.
I don't really need the test now but thought I'd leave it in?
I can remove it now all the data has been fixed as we shouldn't be trying to parse it.

File: test/connectors/httpParsers/VatObligationsHttpParserSpec.scala (hunk 3 above)
Comment (HarryGull) [reply test/connectors/httpParsers/VatObligationsHttpParserSpec.scala]: Absolutely, Why not?
It's an array.
The schema says you can so it is a valid test yes.
Could have more than one.

If we do the code handles it properly and get all the details.
I there are multiple obligations it pulls them out into the

File: test/controllers/VatObligationsControllerSpec.scala (hunk 4 above)
Comment (HarryGull) [reply test/controllers/VatObligationsControllerSpec.scala:40]: Again. Yes according to the schema.
Code handles the situation where it happens so a valid test.

File: test/models/VatObligationsSpec.scala (hunk 5 above)
Comment (HarryGull) [reply test/models/VatObligationsSpec.scala]: No it's correct.
There is a single obligation. 
There are multiple obligation details.
This differentiates it from the case where there are multiple obligations AND multiple obligation details

File: test/models/VatObligationsSpec.scala (hunk 6 above)
Comment (HarryGull) [reply test/models/VatObligationsSpec.scala]: Same reply..

**Pull request merged**